### DISCUSSION TITLE
Howw to querry current workspace /eww

### DISCUSSION THREAD

**Discussion by NicTanghe** [leftwm/leftwm#910] [General]
Hello,

I`m trying to tweak an eww config to work for me on leftwm.

the config I`m trying to adjust is this one.
```
#!/bin/sh
workspaces() {

ws1="1"
ws2="2"
ws3="3"
ws4="4"
ws5="5"
ws6="6"

# Unoccupied
un="0"

# check if Occupied
o1=$(bspc query -D -d .occupied --names | grep "$ws1" )
o2=$(bspc query -D -d .occupied --names | grep "$ws2" )
o3=$(bspc query -D -d .occupied --names | grep "$ws3" )
o4=$(bspc query -D -d .occupied --names | grep "$ws4" )
o5=$(bspc query -D -d .occupied --names | grep "$ws5" )
o6=$(bspc query -D -d .occupied --names | grep "$ws6" )

# check if Focused
f1=$(bspc query -D -d focused --names | grep "$ws1" )
f2=$(bspc query -D -d focused --names | grep "$ws2" )
f3=$(bspc query -D -d focused --names | grep "$ws3" )
f4=$(bspc query -D -d focused --names | grep "$ws4" )
f5=$(bspc query -D -d focused --names | grep "$ws5" )
f6=$(bspc query -D -d focused --names | grep "$ws6" )

ic_1=""
ic_2=""
ic_3=""
ic_4=""
ic_5=""
ic_6=""
if [ $f1 ]; then
    ic_1=""
elif [ $f2 ]; then
    ic_2=""
elif [ $f3 ]; then
    ic_3=""
elif [ $f4 ]; then
    ic_4=""
elif [ $f5 ]; then
    ic_5=""
elif [ $f6 ]; then
    ic_6=""
fi

echo 	"(box	:class \"works\"	:orientation \"h\" :spacing 5 :space-evenly \"false\" (button :onclick \"bspc desktop -f $ws1\"	:class	\"$un$o1$f1\"	\"$ic_1\") (button :onclick \"bspc desktop -f $ws2\"	:class \"$un$o2$f2\"	 \"$ic_2\") (button :onclick \"bspc desktop -f $ws3\"	:class \"$un$o3$f3\" \"$ic_3\") (button :onclick \"bspc desktop -f $ws4\"	:class \"$un$o4$f4\"	\"$ic_4\") (button :onclick \"bspc desktop -f $ws5\"	:class \"$un$o5$f5\" \"$ic_5\")  (button :onclick \"bspc desktop -f $ws6\"	:class \"$un$o6$f6\" \"$ic_6\"))"

}
workspaces
bspc subscribe desktop node_transfer | while read -r _ ; do
workspaces
done
```

There is also the issue that the window’s occupied space is not recognized/ registered and everything is drawn over it.

If anyone is able to help, that would be great.

**Comment by VuiMuich** [comment]
You might want to have a look at the `basic_eww` example you can find [here](https://github.com/leftwm/leftwm/tree/main/themes/basic_eww) with the `yuck` in the `eww_bar` dir.

I guess what you have as `workspaces` in `bspwm` is what we call `tags`, right? Or do you have actually a setup with six displays?

**Comment by NicTanghe** [comment]
hmm when i run those i dont get a rofibar,
I don`t know if im blind but i dont actually c the bar being launched anywhere

Yes those workspaces are tags.

all i c is 
`eww open bar$index`
But index is not assighned inside the script.

**Reply by VuiMuich** [reply]
`index` gets initalized with `0` in line 38 of the `up` script.

Idk, what you mean with "rofibar" tbh.

Did you copy the content of `eww_bar` to `~/.config/eww`?

**Reply by NicTanghe** [reply]
ah i copy`d to .conf/eww/eww-bar

I auto soflink now.

the icon for ram used mem percentage is not in my font though.
I`ll just use 🐏 although i am intreged what should appear

**Reply by VuiMuich** [reply]
The memory symbol requires a "nerdfont". It is a symbol tha very roughly resembles an integrated circuit chip.

**Reply by NicTanghe** [reply]
Where would i tell it to use Cascadia cove ?
I don't c anything about fonts in the .yuck file

**Comment by NicTanghe** [comment]
I've also been looking around for a way to toggle the bar, but I can't seem to find one.

decreicing the heifth in befwindow bar 1 also doesn`t seem to reduce the height of my bar.

**Reply by VuiMuich** [reply]
To toggle a window in `eww` you would use `eww open --toggle <window-name>` in your case `<window-name>` would obviously be `bar1`.

The height of `eww` windows can be extended beyond the configured size by some font glyphs requesting this space. This is essentially a issue of `gtk` so there is not much to do about this besides tinkering with font-sizes.

**Reply by NicTanghe** [reply]
ill have a look at my gtk config then thx.
